src/Extensions/Constants/ShopUrls.php

```diff
@@ -87,6 +103,23 @@ public function orderPropertyFile($path)
         return $this->getShopUrl(RouteConfig::ORDER_PROPERTY_FILE, null, $path);
     }
 
+    public function tracking($orderId)
+    {
+        $lang = $this->sessionStorageService->getLang();
+        return $this->fromMemoryCache("tracking", function() use($orderId, $lang)
+        {
+            $orderRepository = pluginApp(OrderRepositoryContract::class);
+            $orderTrackingService = pluginApp(OrderTrackingService::class);
+            $authHelper = pluginApp(AuthHelper::class);
+            $order = $authHelper->processUnguarded(function() use ($orderRepository, $orderId) {
+                return $orderRepository->findOrderById($orderId);
+            });
+            $trackingURL = $orderTrackingService->getTrackingURL($order, $lang);
```

Comment:
Ich würde den Aufruf von `$orderTrackingService->getTrackingURL($order, $lang)` mit in den processUnguarded-Callback nehmen. Somit wird der Auftrag selbst außerhalb des Callbacks gar nicht zugänglich gemacht.